test/Atmos/EDMF/edmf_model.jl

```diff
@@ -0,0 +1,284 @@
+#### EDMF model
+
+#### Entrainment-Detrainment model
+
+"""
+    EntrainmentDetrainment
+An `EntrainmentDetrainment` model for EDMF, containing
+all related free parameters.
+"""
+Base.@kwdef struct EntrainmentDetrainment{FT <: AbstractFloat}
+    "Entrainmnet TKE scale"
+    c_λ::FT = 0.3
+    "Entrainment factor"
+    c_ε::FT = 0.13
+    "Detrainment factor"
+    c_δ::FT = 0.52
+    "Trubulent Entrainment factor"
+    c_t::FT = 0.1
+    "Detrainment RH power"
+    β::FT = 2
+    "Logistic function scale ‵[1/s]‵"
+    μ_0::FT = 0.0004
+    "Updraft mixing fraction"
+    χ::FT = 0.25
+    "Minimum updraft velocity"
+    w_min::FT = 0.1
+    "Exponential area limiter scale"
+    lim_ϵ::FT = 0.0001
+    "Exponential area limiter amplitude"
+    lim_amp::FT = 10
+    "Minimum Updraft area"
+    a_min::FT = 0.001
+    "Minimum value for turb entr"
+    εt_min::FT = 1e-4
+end
+
+"""
+    SurfaceModel
+A `SurfaceModel` for EDMF, containing all boundary
+values and parameters needed by the model.
+"""
+Base.@kwdef struct SurfaceModel{FT <: AbstractFloat, SV}
+    "Surface temperature ‵[k]‵"
+    surface_T::FT = 300.4
+    "Surface liquid water potential temperature ‵[k]‵"
+    surface_θ_liq::FT = 299.1
+    "Surface specific humidity ‵[kg/kg]‵"
+    surface_q_tot::FT = 22.45e-3
+    "surface sensible heat flux ‵[w/m^2]‵"
+    surface_shf::FT = 9.5
+    "surface lantent heat flux ‵[w/m^2]‵"
+    surface_lhf::FT = 147.2
+    "top sensible heat flux ‵[w/m^2]‵"
+    top_shf::FT = 0.0
+    "top lantent heat flux ‵[w/m^2]‵"
+    top_lhf::FT = 0.0
+    "Sufcae area"
+    a_surf::FT
+    "Sufcae tempearture"
+    T_surf::FT = 300.0
+    "Scalar coefficient"
+    scalar_coeff::SV = 0
+    "Friction velocity"
+    ustar::FT = 0.28
+    "Monin - Obukhov length"
+    obukhov_length::FT = -100
+    "Surface covariance stability coefficient"
+    ψϕ_stab::FT = 8.3
+    "Square ratio of rms turbulent velocity to friction velocity"
+    κ_star²::FT = 3.75
+    "Height of first level"
+    zLL::FT = 60
```

Comment:
same as above

Reply:
AFAIU, this is not a parameter to be tuned. I think the idea is to grab this from the grid.